apps/api/src/auth/index.ts

```diff
@@ -1,10 +1,49 @@
-import type { User } from "@deco/sdk";
+import { Hono } from "hono";
+import {
+  EmailOtpType,
+  type Provider,
+  type User as SupaUser,
+} from "@supabase/supabase-js";
 import { createSupabaseClient } from "../db/client.ts";
-import { getCookies } from "../utils/cookie.ts";
-import { AppContext, getEnv } from "../utils/context.ts";
+import { getCookies, setHeaders } from "../utils/cookie.ts";
+import { AppContext, AUTH_URL, getEnv } from "../utils/context.ts";
+import { authSetCookie, getServerClientOptions } from "../utils/db.ts";
+
+const AUTH_CALLBACK_OAUTH = "/auth/callback/oauth";
+const HAS_LOGGED_IN_COOKIE = "hasLoggedInBefore";
+
+const appAuth = new Hono();
+const appLogin = new Hono();
+export const ROUTES = {
+  ["/auth"]: appAuth,
+  ["/login"]: appLogin,
+} as const;
+
+const createDbAndHeadersForRequest = (ctx: AppContext) => {
+  const { SUPABASE_URL, SUPABASE_SERVER_TOKEN } = getEnv(ctx);
+  const request = ctx.req.raw;
+  const { headers, setCookie } = authSetCookie({ request });
+  const db = createSupabaseClient(
```

Comment:
cant we use the getUser function?

Reply:
You mean, refactor the function getUser to return the user and the database that handles cookies from request?